src/util/ImportUtils.ts

```diff
@@ -39,45 +39,31 @@ export async function importOrRequireFile(
     return tryToRequire()
 }
 
-function getNearestPackageJson(filePath: string): Promise<object | null> {
-    return new Promise((accept) => {
-        let currentPath = filePath
+async function getNearestPackageJson(filePath: string): Promise<object | null> {
+    let currentPath = filePath
 
-        function searchPackageJson() {
-            const nextPath = path.dirname(currentPath)
+    while (currentPath !== path.dirname(currentPath)) {
+        currentPath = path.dirname(currentPath)
+        const potentialPackageJson = path.join(currentPath, "package.json")
 
-            if (currentPath === nextPath)
-                // the top of the file tree is reached
-                accept(null)
-            else {
-                currentPath = nextPath
-                const potentialPackageJson = path.join(
-                    currentPath,
-                    "package.json",
-                )
-
-                fs.stat(potentialPackageJson, (err, stats) => {
-                    if (err != null) searchPackageJson()
-                    else if (stats.isFile()) {
-                        fs.readFile(
-                            potentialPackageJson,
-                            "utf8",
-                            (err, data) => {
-                                if (err != null) accept(null)
-                                else {
-                                    try {
-                                        accept(JSON.parse(data))
-                                    } catch (err) {
-                                        accept(null)
-                                    }
-                                }
-                            },
-                        )
-                    } else searchPackageJson()
-                })
+        try {
+            const stats = await fs.stat(potentialPackageJson)
+            if (stats.isFile()) {
+                try {
+                    return JSON.parse(
+                        await fs.readFile(potentialPackageJson, "utf8"),
+                    )
+                } catch {
+                    return null
+                }
+            } else {
+                continue
             }
+        } catch {
```

Comment:
What are you catching here? A potential throw from fs.stats?

Reply:
The same thing that was caught before: https://nodejs.org/api/errors.html#common-system-errors